File: app/src/main/java/yxd/com/moduledemo/MainActivity.java
package yxd.com.moduledemo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.alibaba.android.arouter.launcher.ARouter;

@Route(path = "/test/main")
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void btn1(View view) {
        Toast.makeText(this, "1", Toast.LENGTH_SHORT).show();
        ARouter.getInstance().build("/test1/module1").navigation();
    }

    public void btn2(View view) {
        ARouter.getInstance().build("/test2/module2").navigation();
    }

    public void btn3(View view) {
        ARouter.getInstance().build("/test/main2").navigation();
    }
}

File: app/src/main/java/yxd/com/moduledemo/MainActivity.kt
package yxd.com.moduledemo

import android.os.Bundle
import com.alibaba.android.arouter.facade.annotation.Route
import com.kotlinlib.activity.AbstractActivity
import com.kotlinlib.new_fragment.FragmentationTabUtils
import com.kotlinlib.other.LayoutId
import com.kotlinlib.other.StatusBarColor
import kotlinx.android.synthetic.main.activity_main.*
import me.yokeyword.fragmentation.SupportFragment
import yxd.com.module1.Module1Fragment
import yxd.com.module2.ModuleFragment2

@StatusBarColor("#66ff66")
@LayoutId(R.layout.activity_main)
@Route(path = "/test/main")
class MainActivity : AbstractActivity() {

    override fun init(bundle: Bundle?) {

        FragmentationTabUtils<SupportFragment>(
                this,
                tabLayout,
                R.id.flContainer,
                {
                    i, tab ->
                    return@FragmentationTabUtils tab.setText("Module${i+1}")
                },
                {
                    fu, frags, tab ->
                    fu.switch(tab.position)
                },
                {
                    tab ->
                },
                Module1Fragment(), ModuleFragment2()
        )

        tabLayout.setSelectedTabIndicatorHeight(0)

    }

}
